### PR TITLE
Support Debian 12 for default image

### PR DESCRIPTION
**Pull request by k-304**
Support tests with Debian 12 as default image.

### CODE REVIEW

File: test_private_network.py [hunk 1]
```diff
@@ -142,33 +142,75 @@ def get_interface_config(server_index):
...
     """
 
+    # Define a custom subnet
+    subnet = private_network.add_subnet(cidr='192.168.100.0/24')
+
+    s1_address = '192.168.100.1'
+    s2_address = '192.168.100.1'
+
+    # Initialize each server with an interface attached to the private network.
+    # The servers are not given an address by DHCP
```
Comment (href) [inline test_private_network.py:165]: I think the way we write this in the API docs is a tad clearer:

> No IP address will be assigned using DHCP

https://www.cloudscale.ch/en/api/v1#interfaces-attribute-specification

File: test_private_network.py [hunk 2]
```diff
@@ -142,33 +142,75 @@ def get_interface_config(server_index):
...
             'name': 's2',
             'image': image,
             'use_private_network': True,
             'server_groups': [server_group.uuid],
+            'interfaces': interfaces,
         },
     ))
 
-    # Change the MAC addresses of the private interfaces
+    # Make sure the address is actually unset
```
Comment (href) [inline test_private_network.py:189]: To be more in line with the above:

> Make sure no IP address was assigned using DHCP

File: test_private_network.py [hunk 3]
```diff
@@ -142,33 +142,75 @@ def get_interface_config(server_index):
...
+    assert s1.ip('private', 4, fail_if_missing=False) is None
+    assert s2.ip('private', 4, fail_if_missing=False) is None
+
+    # Make sure the host did not get an address matching the subnet
```
Comment (href) [inline test_private_network.py:193]: > Make sure the host was not assigned an address matching the subnet

File: test_public_network.py [hunk 4]
```diff
@@ -67,7 +67,7 @@ def test_public_network_connectivity_on_all_images(server):
         server.ping(address, count=3, interval=0.5)
 
 
-def test_public_network_mtu(server):
+def test_public_network_mtu(server, image):
```
Comment (href) [inline test_public_network.py:70]: Why has `image` been added here? It seems unused.

**Review by href — COMMENTED** [review]
See comments. The approach is sound and well-implemented, I have nothing but nitpicks.

File: test_private_network.py (hunk 1 above)
Comment (k-304) [reply test_private_network.py:165]: Updated with text from API.

File: test_private_network.py (hunk 3 above)
Comment (k-304) [reply test_private_network.py:193]: Updated with suggested text.

File: test_public_network.py (hunk 4 above)
Comment (k-304) [reply test_public_network.py:70]: That seems to be a leftover, removed.

**Review by href — APPROVED** [review]

**Pull request merged**